lib/hooks.js

```diff
@@ -62,9 +69,45 @@ function buildHooks (h) {
   hooks.onError = h.onError.slice()
   hooks.onRoute = h.onRoute.slice()
   hooks.onRegister = h.onRegister.slice()
+  hooks.onCompilerReady = []
```

Comment:
I think that this hook should trigger once so I didn't use `slice()`

Otherwise, the hook will be called for every `register` the user will do